contracts/VaultOUSDLvUSD.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+/*
+  Vault is just a simple in/out it provide a fixed (non upgradable) address to transfer OUSD and lvUSD to (and pay interest into).
+  That means we need to change the architecture and add Minter or have the coordinator deal with than
+*/
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "hardhat/console.sol";
+
+contract VaultOUSDLvUSD {
+    address private oUSDContractAddress;
+    address private lvUSDContractAddress;
+
+    uint256 private oUSDRebasedBalance;
+    uint256 private lvUSDBalance;
+
+    constructor(address oUSDContract, address lvUSDContract) {
```

Comment:
Since I can't necessarily verify the functionality 100% today I'm focusing on nitpicky stuff... In online tutorials I've seen examples use underscored versions of the variable names as the parameters. For me, it makes it easier to read and understand.

```
constructor(address _oUSDContractAddress, address _lvUSDContractAddress) {
    oUSDContractAddress = _oUSDContractAddress;
    oUSDContractAddress = _lvUSDContractAddress;
    ...
}
```